Sources/App/CRP/CRPProcess.swift

```diff
@@ -0,0 +1,122 @@
+import Foundation
+import Vapor
+import Stevenson
+
+enum CRPProcess {
+    enum Option: String {
+        case repo
+        case branch
+        case skipTicket
+        case skipFixVersion
+
+        func get<T: Decodable>(from request: Request) throws -> T {
+            guard let result = request.query[T.self, at: self.rawValue] else {
+                throw CRPProcess.Error.missingParameter(self.rawValue)
+            }
+            return result
+        }
+    }
+
+    enum Error: Swift.Error {
+        case invalidParameter(key: String, value: String, expected: String)
+        case missingParameter(String)
+    }
+
+    static func apiRequest(request: Request, github: GitHubService, jira: JiraService) throws -> Future<Response> {
+        let repo: String = try Option.repo.get(from: request)
+        let branch: String = try Option.branch.get(from: request)
+        let skipTicket: Bool = (try? Option.skipTicket.get(from: request)) ?? false
+        let skipFixVersion: Bool = (try? Option.skipFixVersion.get(from: request)) ?? false
+
+        guard let repoMapping = RepoMapping.all[repo.lowercased()] else {
+            throw CRPProcess.Error.invalidParameter(
+                key: Option.repo.rawValue,
+                value: repo,
+                expected: RepoMapping.all.keys.joined(separator: "|")
+            )
+        }
+
+        let release = try GitHubService.Release(
+            repo: repoMapping.repository,
+            branch: branch
+        )
+
+        return try github.changelog(for: release, on: request)
+            .catchError(.capture())
+            .flatMap { (commitMessages: [String]) -> Future<(JiraService.CreatedIssue?, JiraService.FixedVersionReport?)> in
+
+                let jiraVersionName = repoMapping.crp.jiraVersionName(release)
+                let changelogSections = ChangelogSection.makeSections(from: commitMessages, for: release)
+
+                // Create CRP ticket, unless skipped
+                let createTicket: Future<JiraService.CreatedIssue?>
+                if skipTicket {
+                    createTicket = request.future(nil)
+                } else {
+                    // Create CRP Issue
+                    let crpIssue = JiraService.makeCRPIssue(
+                        jiraBaseURL: jira.baseURL,
+                        crpProjectID: JiraService.crpProjectID,
+                        crpConfig: repoMapping.crp,
+                        release: release,
+                        changelog: jira.document(from: changelogSections)
+                    )
+
+                    createTicket = try jira.create(issue: crpIssue, on: request)
+                        .map { Optional.some($0) }
```

Comment:
Isn't it similar to `.map(Optional.some)`? 🤔

Reply:
indeed :+1: